app/models/package_manager/base.rb

```diff
@@ -99,8 +100,11 @@ def self.save(project)
       end
 
       if self::HAS_VERSIONS
+        class_name = name.demodulize
         versions(project, dbproject.name).each do |version|
-          dbproject.versions.create(version) unless dbproject.versions.find { |v| v.number == version[:number] }
+          existing = dbproject.versions.find_or_initialize_by(number: version[:number])
+          existing.repository_sources = Set.new(existing.repository_sources).add(class_name).to_a if self::HAS_MULTIPLE_REPO_SOURCES
```

Comment:
This looks good. I'm wondering if this still means we gotta backfill maven, or maybe we'll just handle for it once we're looking at it. In any case, we can do it later.